lib/client/rpcClient.ts

```diff
@@ -0,0 +1,46 @@
+import { Program, Provider, utils } from "@coral-xyz/anchor";
+import { AutocratProgram, Dao, DaoState, ProgramVersion } from "../types";
+import { PublicKey } from "@solana/web3.js";
+import { MetaDAOClient } from "./client";
+import { enrichTokenMetadata } from "../tokens";
+
+export class MetaDAORPClient implements MetaDAOClient {
+  private programVersion: ProgramVersion;
+  private rpcProvider: Provider;
+  private constructor(programVersion: ProgramVersion, rpcProvider: Provider) {
+    this.programVersion = programVersion;
+    this.rpcProvider = rpcProvider;
+  }
+  static make(programVersion: ProgramVersion, rpcProvider: Provider) {
+    return new MetaDAORPClient(programVersion, rpcProvider);
+  }
+  async fetchDao(daoAddress: string): Promise<Dao | undefined> {
+    const { programId, idl } = this.programVersion!;
+
+    const dao = PublicKey.findProgramAddressSync(
+      [utils.bytes.utf8.encode(daoAddress)],
```

Comment:
Starting in V0.3 we'll need to fetch DAOs through getProgramAccounts where as before we had the seed to derive off of, which we still don't have a hard fix on, if we'd prefer... I think without seed is more flexible with future proofing eg. `jupD4o2LAfJYbRMAHHvqjCwQxanZn7ReEHp1k81EohpZb` so people can pass in their custom addresses. So the OLD way of doing it will no longer work so a check for the version or `isMultiDAO` (see below) is likely ideal for backwards compatibility.

Reply:
Using the `.all()` method now per our discussion so this should be  good.